src/libsync/syncengine.cpp

```diff
@@ -652,6 +652,11 @@ void SyncEngine::slotDiscoveryFinished()
     //    qCInfo(lcEngine) << "Permissions of the root folder: " << _csync_ctx->remote.root_perms.toString();
     auto finish = [this]{
         auto databaseFingerprint = _journal->dataFingerprint();
+        // fix issue in the database -> we shouldn't have stored "[empty]" as now it leads to always restoring the files even though the fingerprint hasn't really changed
```

Comment:
Then following up my train of thought from the previous comment, isn't the mistake that we just look for `!databaseFingerprint.isEmpty()` later on? Shouldn't we have something like:
`const auto emptyFingerprint = databaseFingerprint.isEmpty() || databaseFingerprint == QByteArrayLiteral("[empty]");`
and then use that in the if?

This would better align with the comment which says "for example, upgrading from a previous version, or first sync," while currently checking only with `isEmpty()` satisfies only the "or server not supporting fingerprint" case.